torch/optim/sgd.py

```diff
@@ -46,7 +46,7 @@ def step(self, closure=None):
                 if weight_decay != 0:
                     d_p.add_(weight_decay, p.data)
                 if momentum != 0:
-                    param_state = self.state[id(p)]
+                    param_state = self.state[p]
```

Comment:
breaks a lot of my code eg here https://github.com/szagoruyko/attention-transfer/blob/master/cifar.py#L314 but ok, my fault for not resetting the optimizer..